### PR TITLE
evm contract init and relayer support

### PR DESCRIPTION
**Pull request by dixitaniket**
- adds evm support
- evm smart contract  with logic similar to cosmwasm contracts
- evm tests and relayer unit tests

closes https://github.com/ojo-network/contracts/issues/89

### CODE REVIEW

File: cw-relayer/config/config.go [hunk 1]
```diff
@@ -137,6 +135,10 @@ func ParseConfig(configPath string) (Config, error) {
 		cfg.MaxTickTimeout = defaultTimeout.String()
 	}
 
+	if cfg.Keyring.PrivKey == "$PRIV_KEY" {
+		cfg.Keyring.PrivKey = key
+	}
```
Comment (adamewozniak) [inline cw-relayer/config/config.go:140]: Can nested objects not be set via environment variables with viper?
I believe @zarazan might have done something on this

File: cw-relayer/go.mod [hunk 2]
```diff
@@ -3,13 +3,11 @@ module github.com/ojo-network/cw-relayer
 go 1.19
 
 require (
-	github.com/CosmWasm/wasmd v0.0.0-00010101000000-000000000000
 	github.com/cosmos/cosmos-sdk v0.46.7
-	github.com/cosmos/go-bip39 v1.0.0
+	github.com/ethereum/go-ethereum v1.10.17
```
Comment (adamewozniak) [inline cw-relayer/go.mod:7]: We're going to need dependency update workflows directly to this branch

File: cw-relayer/relayer/client/client.go [hunk 3]
```diff
@@ -1,178 +1,168 @@
...
-	n, err = r.buf.Read(p)
-	if err == io.EOF || n == 0 {
-		r.buf.WriteString(r.pass + "\n")
+	g, _ := errgroup.WithContext(ctx)
 
-		n, err = r.buf.Read(p)
-	}
+	var mut sync.Mutex
+	var response QueryResponse
+	asset := tools.StringToByte32(assetName)
```
Comment (adamewozniak) [inline cw-relayer/relayer/client/client.go:114]: should we put this in a helper function to make it more descriptive?

File: cw-relayer/relayer/client/client.go [hunk 4]
```diff
@@ -1,178 +1,168 @@
...
+	relayerClient.client = ethClient
 
 	return relayerClient, nil
 }
 
-func newPassReader(pass string) io.Reader {
-	return &passReader{
-		pass: pass,
-		buf:  new(bytes.Buffer),
+func (oc RelayerClient) BroadcastContractQuery(ctx context.Context, assetName string) (QueryResponse, error) {
```
Comment (adamewozniak) [inline cw-relayer/relayer/client/client.go:100]: Let's add a godoc here with each step in the function

File: cw-relayer/relayer/client/client.go [hunk 5]
```diff
@@ -1,178 +1,168 @@
...
 
-	return n, err
+	err = g.Wait()
+	return response, err
 }
 
 // BroadcastTx attempts to broadcast a signed transaction. If it fails, a few re-attempts
 // will be made until the transaction succeeds or ultimately times out or fails.
-func (oc RelayerClient) BroadcastTx(nextBlockHeight, timeoutHeight int64, msgs ...sdk.Msg) error {
+func (oc RelayerClient) BroadcastTx(nextBlockHeight, timeoutHeight uint64, rate []PriceFeedData, deviation []PriceFeedData, medians []PriceFeedMedianData, disableResolve bool) error {
```
Comment (adamewozniak) [inline cw-relayer/relayer/client/client.go:159]:
```suggestion
func (oc RelayerClient) BroadcastTx(nextBlockHeight, timeoutHeight uint64, rates []PriceFeedData, deviations []PriceFeedData, medians []PriceFeedMedianData, disableResolve bool) error {
```

File: cw-relayer/relayer/messages.go [hunk 6]
```diff
@@ -1,173 +1,60 @@
...
-		Relay Msg `json:"force_relay"`
-	}
-
-	MsgRelayHistoricalMedian struct {
-		Relay Msg `json:"relay_historical_median"`
-	}
-
-	MsgForceRelayHistoricalMedian struct {
-		Relay Msg `json:"force_relay_historical_median"`
+func (r *Relayer) genRateMsgs(requestID uint64, resolveTime uint64) (msg []client.PriceFeedData) {
```
Comment (adamewozniak) [inline cw-relayer/relayer/messages.go:12]: add a godoc here please - `genRateMsgs is a function which converts x to y for consumption by z`

File: evm/README.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+Try running some of the following tasks:
+
+```shell
+npx hardhat help
+npx hardhat test
+REPORT_GAS=true npx hardhat test
+npx hardhat node
+npx hardhat run scripts/deploy.ts
+```
```
Comment (adamewozniak) [inline evm/README.md:13]: Let's update this please 🙏

**Review by adamewozniak — APPROVED** [review]
utACK - looks good!

Let's add a section to the main branch's readme about where this lives too

File: cw-relayer/config.toml [hunk 8]
```diff
@@ -42,16 +42,12 @@ skip_error = true
...
-address = "wasm1usr9g5a4s2qrwl63sdjtrs2qd4a7huh6qksawp"
-chain_id = "wasm-test"
-acc_prefix = "wasm"
+address = "0x292505757b2380705231Fb69f964f05Aebc078E6"
+chain_id = "80001"
 ### keyring for the relayer account on the wasmd chain
 [keyring]
-backend = "test"
-dir = "./"
+priv_key = "$PRIV_KEY"
```
[the remark this answers is not in the document]
Comment (dixitaniket) [reply cw-relayer/config.toml:49]: yes, either this or one can use it without the env variable as well, currently looking into a sort of keystore solution

File: cw-relayer/go.mod (hunk 2 above)
Comment (dixitaniket) [reply cw-relayer/go.mod:7]: would modify depandabot config to make updates against evm branch

**Pull request merged**